prime-dotnet-webapi/Controllers/VerifiableCredentialController.cs

```diff
@@ -14,29 +16,39 @@ namespace Prime.Controllers
     public class VerifiableCredentialController : ControllerBase
     {
         private readonly IVerifiableCredentialService _verifiableCredentialsService;
+        private readonly ILogger _logger;
 
         public VerifiableCredentialController(
-            IVerifiableCredentialService verifiableCredentialService
+            IVerifiableCredentialService verifiableCredentialService,
+            ILogger<VerifiableCredentialController> logger
         )
         {
             _verifiableCredentialsService = verifiableCredentialService;
+            _logger = logger;
         }
 
-        // POST: api/topic/:topic
+        // POST: api/topic/:topic/{guid}
         /// <summary>
         /// Handle webhook events sent from the issuing agent.
         /// </summary>
+        /// <param name="apiKey"></param>
         /// <param name="topic"></param>
         /// <param name="data"></param>
         // Webhooks explained "In Soviet Russia API call you!" - Jason Aitchison 2020
-        // TODO add security to the webhook receiver endpoint
-        [HttpPost("/api/webhooks/topic/{topic}", Name = nameof(Webhook))]
+        [HttpPost("/api/webhooks/{apiKey}/topic/{topic}", Name = nameof(Webhook))]
         [ProducesResponseType(typeof(ApiBadRequestResponse), StatusCodes.Status400BadRequest)]
         [ProducesResponseType(StatusCodes.Status401Unauthorized)]
         // TODO update to response code 202 when queue has been added for webhooks
         [ProducesResponseType(StatusCodes.Status204NoContent)]
-        public async Task<ActionResult> Webhook(string topic, [FromBody] JObject data)
+        public async Task<ActionResult> Webhook(string apiKey, string topic, [FromBody] JObject data)
         {
+            _logger.LogInformation($"API-KEY: [{apiKey}], VERIFIABLE_CREDENTIAL_WEBHOOK_KEY: [{PrimeConstants.VERIFIABLE_CREDENTIAL_WEBHOOK_KEY}]");
+
+            if (apiKey != PrimeConstants.VERIFIABLE_CREDENTIAL_WEBHOOK_KEY.Trim())
```

Comment:
Is the use of `Trim()` on a PrimeConstant a test to see if there was an issue?

Reply:
AJ couldnt figure out why there was a newline character in the env variable